### PR TITLE
nodePackages.prisma: 5.0.0 -> 5.2.0

### PR DESCRIPTION
**Pull request by aqrln**
## Description of changes

Update `prisma-engines` and `nodePackages.prisma` to 5.2.0

* https://github.com/prisma/prisma/releases/tag/5.1.0
* https://github.com/prisma/prisma/releases/tag/5.1.1
* https://github.com/prisma/prisma/releases/tag/5.2.0

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc maintainers @tomhoule @ivan

### CODE REVIEW

**Comment by aqrln** [comment]
@SuperSandro2000 you previously said that we should eventually move this out of `nodePackages`, where would it be better to move it instead? Is there any example I could take a look at?

File: pkgs/development/node-packages/overrides.nix [hunk 1]
```diff
@@ -367,12 +367,11 @@ final: prev: {
 
     src = fetchurl {
       url = "https://registry.npmjs.org/prisma/-/prisma-${version}.tgz";
-      hash = "sha256-0NxYp+W2KbR3xEV2OCXCIL3RqkvLfJHNKgl/PxapVbI=";
+      hash = "sha256-gikKT2iUdSb7/qOpJNiukbm35ggLcI2Y24uuGIqpSbE=";
     };
     postInstall = with pkgs; ''
       wrapProgram "$out/bin/prisma" \
         --set PRISMA_SCHEMA_ENGINE_BINARY ${prisma-engines}/bin/schema-engine \
-        --set PRISMA_MIGRATION_ENGINE_BINARY ${prisma-engines}/bin/schema-engine \
```
Comment (aqrln) [inline pkgs/development/node-packages/overrides.nix:375]: Deprecated env var, not necessary anymore (it's now covered by `PRISMA_SCHEMA_ENGINE_BINARY` above).

**Comment by aqrln** [comment]
Result of `nixpkgs-review pr 249148` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>prisma-engines</li>
  </ul>
</details>

**Review by Jolg42 — APPROVED** [review]

**Review by tomhoule — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
> you previously said that we should eventually move this out of `nodePackages`

Better sooner than later because it makes literally everything easier.

> where would it be better to move it instead?

Somewhere fitting like any other package under pkgs.

> Is there any example I could take a look at?

Take a look at the other packages which migrated out of nodePackages.

**Comment by sarahec** [comment]
> [v5.2](https://github.com/prisma/prisma/releases) was released a couple of days ago. Should we just skip right to it?

I was getting ready to make a PR for that, but it sounds like I should defer to you all.

**Comment by aqrln** [comment]
@SuperSandro2000 I'm sorry but I couldn't figure out how to do this, and I could not find any relevant examples of other pnpm-based projects that were moved out of `nodePackages`. All packages I saw used `buildNpmPackage` + `fetchFromGitHub` but that's not currently possible for Prisma because the necessary tooling to reproducibly build it from source in nixpkgs doesn't exist yet.

I tried using `buildNpmPackage` by fetching the released tarball from the npm registry as `src` and setting `dontNpmBuild = true`. The idea was to to provide a fake `package-lock.json` file that only contained the single runtime dependency of the `prisma` package and copy it in `postPatch`, but that didn't work as I still got `ENOTCACHED` errors from npm for the devDependencies, even with `﻿﻿﻿npmInstallFlags = [ "--omit" "dev" ]`. It would also add more maintenance cost even if it worked since that custom lockfile would need to be updated somehow (either manually or with some script).

I'd suggest we keep it as is for now. If you have any ideas or suggestions about how to approach this, I'm happy to try that in a separate PR.

**Comment by tomhoule** [comment]
Seconding @aqrln on there not being a good nix way to build pnpm projects that I am aware of. Unfortunately, pnpm is different enough from npm.

**Review by donovanglover — APPROVED** [review]

**Pull request merged**